src/ocean/core/VersionCheck.d

```diff
@@ -0,0 +1,28 @@
+module ocean.core.VersionCheck;
+
+import ocean.transition;
+import ocean.core.Traits;
+
+public template hasFeaturesFrom ( istring libname, ulong major, ulong minor )
+{
+    const hasFeaturesFrom = hasFeaturesFromFunc!(libname, major, minor)();
+}
+
+private bool hasFeaturesFromFunc ( istring libname, ulong major, ulong minor ) ()
+{
+    struct Library
+    {
+        mixin("import " ~ libname ~ ".LibVersion;");
+    }
+
+    mixin("return is(typeof(Library.has_features_"
+        ~ ctfe_i2a(major) ~ "_" ~ ctfe_i2a(minor) ~ "));");
```

Comment:
Same as usual. Library usage of `-de` does not propagate to applications - deprecating symbol will ensure it is not directly used in libraries but will keep working if app upgrades to ocean version above supported to library. Which is exactly what we want, right?